### DISCUSSION TITLE
Is it normal for useEffect(..., []) to run twice?

### DISCUSSION THREAD

**Discussion by wunksert** [remix-run/remix#7096] [Q&A] [answered]
I'm seeing `useEffect` run twice when a route renders. Set it up with an empty dependency array and a `console.log` and it logs twice to the browser console. If this is normal, what's the cause?

### Reproduce
Using a clean Indie stack template

in `routes/myroute.jsx`

```
export default function MyRoute(){

    useEffect(() => {
        console.log("In useEffect");
    }, [])

    return(
        <p>hi</p>
    )
}
```

output:
```
In useEffect
In useEffect
```

**Comment by sergiodxa** [comment, marked as answer]
This is happening because on your app is wrapped in `React.StrictMode` inside the entry.client file.
